value_test.go

```diff
@@ -935,3 +935,51 @@ func BenchmarkReadWrite(b *testing.B) {
 		}
 	}
 }
+
+// Regression test for https://github.com/dgraph-io/badger/issues/817
+func TestValueLogTruncate(t *testing.T) {
+	dir, err := ioutil.TempDir(".", "badger-test")
+	y.Check(err)
+	defer os.RemoveAll(dir)
+
+	opts := DefaultOptions
+	opts.Dir = dir
+	opts.ValueDir = dir
+	opts.Truncate = true
+
+	db, err := Open(opts)
+	require.NoError(t, err)
+	// Insert 1 entry so that we have valid data in first vlog file
+	require.NoError(t, db.Update(func(txn *Txn) error {
+		require.NoError(t, txn.Set([]byte("foo"), nil))
+		return nil
+	}))
+
+	fileCountBeforeCorruption := len(db.vlog.filesMap)
+	require.NoError(t, err)
+	require.NoError(t, db.Close())
+	// Create two vlog files corrupted data. These will be truncated when DB starts next time
+	require.NoError(t, ioutil.WriteFile(vlogFilePath(dir, 1), []byte("foo"), 0664))
+	require.NoError(t, ioutil.WriteFile(vlogFilePath(dir, 2), []byte("foo"), 0664))
+
+	db, err = Open(opts)
+	require.NoError(t, err)
+
+	// Ensure vlog file with id=1 is not present
+	require.Nil(t, db.vlog.filesMap[1])
+
+	// Ensure filesize of fid=2 is zero
+	zeroFile, ok := db.vlog.filesMap[2]
+	require.True(t, ok)
+	fileStat, err := zeroFile.fd.Stat()
+	require.NoError(t, err)
+	require.Zero(t, fileStat.Size())
+
+	// -1 because the file with id=2 will be completely truncated. It won't be deleted.
+	// There would be two files. fid=0 with valid data, fid=2 with zero data (truncated).
+	fileCountAfterCorruption := len(db.vlog.filesMap) - 1
+	require.Equal(t, fileCountBeforeCorruption, fileCountAfterCorruption)
```

Comment:
```suggestion
	require.Equal(t, fileCountBeforeCorruption+1, fileCountAfterCorruption)
```
is more clear to understand. Instead of comparing fileCountBeforeCorruption to fileCountAfterCorruption-1

Reply:
Thanks! Changed it.